src/dataflow/src/sink/kafka.rs

```diff
@@ -216,14 +216,22 @@ where
                 };
 
                 // Grab all of the available Rows and put them in a queue before we
-                // send it over to Kafka. We Even though we want to do bounded work
+                // send it over to Kafka. Even though we want to do bounded work
                 // per sink invocation, we still need to remember all inputs as we
-                // receive them
+                // receive them.
                 input.for_each(|_, rows| {
                     rows.swap(&mut vector);
 
-                    for (row, _time, diff) in vector.drain(..) {
-                        queue.push_back((row, diff));
+                    for (row, time, diff) in vector.drain(..) {
+                        let should_emit = if connector.strict {
```

Comment:
I think it might be worth it to throw this logic in a function in src/dataflow/src/sink/util.rs now that we're repeating it enough times and likely will want to repeat it in the future too? caveat here that I don't have a great sense of how hard making a new function will be given rust / type system fun so if it turns out to be a giant rabbit hole I'm sorry in advance